misk/src/main/kotlin/misk/hibernate/HibernateHealthCheck.kt

```diff
@@ -0,0 +1,37 @@
+package misk.hibernate
+
+import misk.healthchecks.HealthCheck
+import misk.healthchecks.HealthStatus
+import misk.jdbc.DataSourceConfig
+import org.hibernate.SessionFactory
+import java.sql.Timestamp
+import java.time.Clock
+import java.time.Duration
+
+/**
+ * HealthCheck to confirm database connectivity and defend against clock skew.
+ */
+class HibernateHealthCheck(
+  private val sessionFactory: SessionFactory,
+  private val config: DataSourceConfig,
+  private val clock: Clock
+) : HealthCheck {
+
+  override fun status(): HealthStatus {
+    val databaseInstant = try {
+      sessionFactory.openSession().use { session ->
+        session.createNativeQuery("SELECT NOW()").uniqueResult() as Timestamp
+      }.toInstant()
+    } catch (e: Exception) {
+      return HealthStatus.unhealthy("Hibernate: failed to query ${config.database}")
+    }
+
+    val delta = Duration.between(clock.instant(), databaseInstant).abs()
+
+    return if (delta > config.clock_skew_threshold) {
+      HealthStatus.unhealthy("Hibernate: host clock has drifted ${delta.seconds}s from ${config.database}")
```

Comment:
we can’t _really_ blame the host; all we know is that the host and one of its DBs disagree.

Reply:
Fixed language to make that more clear.